tuf/repository_lib.py

```diff
@@ -1506,9 +1523,27 @@ def generate_snapshot_metadata(metadata_directory, version, expiration_date,
       # list these roles found in the metadata directory.
       if tuf.roledb.role_exists(rolename, repository_name) and \
           rolename not in ['root', 'snapshot', 'timestamp', 'targets']:
-        fileinfodict[metadata_name] = get_metadata_versioninfo(rolename,
+
+        length = None
+        hashes = None
+        if use_length or use_hashes:
+
+          length, hashes = securesystemslib.util.get_file_details(
+            os.path.join(metadata_directory, metadata_filename),
+            tuf.settings.FILE_HASH_ALGORITHMS)
+
+          if use_length == False:
```

Comment:
Maybe replace this with a conditional assignment like:

`length = (use_length and length) or None`

Reply:
I rebased and addressed that comment.